flask/helpers.py

```diff
@@ -534,8 +535,13 @@ def send_file(filename_or_fp, mimetype=None, as_attachment=False,
         if attachment_filename is None:
             raise TypeError('filename unavailable, required for '
                             'sending as attachment')
+        filename_dict = {
+            'filename': (unicodedata.normalize('NFKD',
+                         text_type(attachment_filename)).encode('ascii',
+                                                                'ignore')),
+            'filename*': "UTF-8''%s" % url_quote(attachment_filename)}
```

Comment:
We need to ensure that the order here is retained, since IE8 breaks if `filename*` is first in the header. Since we're unpacking the dict, we can't ensure order. One solution is to patch Werkzeug to allow `*args` to `headers.add`.